### PR TITLE
Support for large file uploads to Azure Storage

### PR DESCRIPTION
**Pull request by iamrz1**
xref: wal-g#74

### CODE REVIEW

**Comment by iamrz1** [comment]
Ignore this PR.

**Comment by iamrz1** [comment]
@x4m PTAL

File: README.md [hunk 1]
```diff
@@ -49,6 +49,22 @@ To store backups in Google Cloud Storage, WAL-G requires that this variable be s
 WAL-G determines Google Cloud credentials using [application-default credentials](https://cloud.google.com/docs/authentication/production) like other GCP tools. You can set `GOOGLE_APPLICATION_CREDENTIALS` to point to a service account json key from GCP. If you set nothing, WAL-G will attempt to fetch credentials from the GCE/GKE metadata service.
 
 
+To store backups in Azure Storage, WAL-G requires that this variable be set:
+
+* `WALG_AZ_PREFIX` to specify where to store backups in Azure storage (eg. `azure://x4m-test-bucket/walg-folder`)
+
+WAL-G determines Azure Storage credentials using [azure default credentials](https://docs.microsoft.com/en-us/azure/storage/common/storage-azure-cli#azure-cli-sample-script). You can set `AZURE_STORAGE_ACCOUNT`, `AZURE_STORAGE_ACCESS_KEY` to provide azure storage credentials.
+
+Use `WALG_AZURE_BUFFER_SIZE` to set the upload buffer size, and `WALG_AZURE_MAX_BUFFERS` to set the number of buffers to use at most. The default buffer size is  64 * 1024 * 1024 bytes, and the default number of buffers is 3.
```
Comment (g0djan) [inline README.md]: Write a description of WALG_AZURE_BUFFER_SIZE and WALG_AZURE_MAX_BUFFERS in the same style of the document. 

Purpose of this variable:
* `NAME_OF_VARIABLE` - some description with examples.

File: internal/storages/azure/folder.go [hunk 2]
```diff
@@ -178,3 +172,20 @@ func (folder *Folder) DeleteObjects(objectRelativePaths []string) error {
...
+	if err != nil || bufferSize < 1024 {
+		bufferSize = 64 * 1024 * 1024
+	}
+	// Configure the size of the rotating buffers and number of buffers
+	maxBufS := os.Getenv("WALG_AZURE_MAX_BUFFERS")
+	maxBuffers, err := strconv.Atoi(maxBufS)
+	if err != nil || maxBuffers < 1 {
+		maxBuffers = 3
+	}
+	uploadOptions := azblob.UploadStreamToBlockBlobOptions{MaxBuffers: maxBuffers, BufferSize: bufferSize}
```
Comment (g0djan) [inline internal/storages/azure/folder.go]: Creating a redundant variable. Just return it.

File: internal/storages/azure/folder.go [hunk 3]
```diff
@@ -178,3 +172,20 @@ func (folder *Folder) DeleteObjects(objectRelativePaths []string) error {
 	}
 	return nil
 }
+
+func getUploadStreamToBlockBlobOptions() azblob.UploadStreamToBlockBlobOptions {
+	// Configure the size of the rotating buffers
+	bufSizeS := os.Getenv("WALG_AZURE_BUFFER_SIZE")
+	bufferSize, err := strconv.Atoi(bufSizeS)
+	if err != nil || bufferSize < 1024 {
```
Comment (g0djan) [inline internal/storages/azure/folder.go]: Replace this magic number with a symbolic constant.

File: internal/storages/azure/folder.go [hunk 4]
```diff
@@ -178,3 +172,20 @@ func (folder *Folder) DeleteObjects(objectRelativePaths []string) error {
...
+		bufferSize = 64 * 1024 * 1024
```
Comment (g0djan) [inline internal/storages/azure/folder.go]: Replace this magic numbers with symbolic constants.

File: internal/storages/azure/folder.go [hunk 5]
```diff
@@ -178,3 +172,20 @@ func (folder *Folder) DeleteObjects(objectRelativePaths []string) error {
...
+	}
+	// Configure the size of the rotating buffers and number of buffers
+	maxBufS := os.Getenv("WALG_AZURE_MAX_BUFFERS")
+	maxBuffers, err := strconv.Atoi(maxBufS)
+	if err != nil || maxBuffers < 1 {
```
Comment (g0djan) [inline internal/storages/azure/folder.go]: Replace this magic number with a symbolic constant.

File: internal/storages/azure/folder.go [hunk 6]
```diff
@@ -178,3 +172,20 @@ func (folder *Folder) DeleteObjects(objectRelativePaths []string) error {
...
+		maxBuffers = 3
```
Comment (g0djan) [inline internal/storages/azure/folder.go]: Replace this magic number with a symbolic constant.

**Comment by iamrz1** [comment]
@g0djan Made requested changes

**Pull request merged**